wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift

```diff
@@ -934,66 +995,72 @@ extension ZMUserSession: ZMSyncStateDelegate {
             ).post()
         }
 
-        let selfClient = ZMUser.selfUser(in: syncContext).selfClient()
-
-        if selfClient?.hasRegisteredMLSClient == true {
-            Task {
-                do {
-                    try await mlsService.repairOutOfSyncConversations()
-                } catch {
-                    WireLogger.mls.error("Repairing out of sync conversations failed: \(error)")
+        syncContext.perform { [weak self] in
```

Comment:
Before we (bravely) assumed that this method would be invoked on the sync context, now it's not always true, so best to switch to it explicitly.